src/vtol.py

```diff
@@ -184,9 +157,71 @@ def change_status(self, new_status):
             raise Exception("Error: Unsupported status for vehicle")
         self.status = new_status
 
-    def include_heading(self):
-        '''Includes heading in messages'''
-        self.heading = True
+
+    def send_attitude_target(
+            self,
+            roll_angle=0.0,
+            pitch_angle=0.0,
+            yaw_angle=None,
+            yaw_rate=0.0,
+            use_yaw_rate=False,
+            thrust=0.5,
+    ):
+        '''
+        use_yaw_rate: the yaw can be controlled using yaw_angle OR yaw_rate.
+                    When one is used, the other is ignored by Ardupilot.
+        thrust: 0 <= thrust <= 1, as a fraction of maximum vertical thrust.
+                Note that as of Copter 3.5, thrust = 0.5 triggers a special case in
+                the code for maintaining current altitude.
+        '''
+        if yaw_angle is None:
+            # this value may be unused by the vehicle, depending on use_yaw_rate
+            yaw_angle = self.attitude.yaw
+        # Thrust >  0.5: Ascend
+        # Thrust == 0.5: Hold the altitude
+        # Thrust <  0.5: Descend
+        msg = self.message_factory.set_attitude_target_encode(
+            0,  # time_boot_ms
+            1,  # Target system
+            1,  # Target component
+            0b00000000 if use_yaw_rate else 0b00000100,
+            to_quaternion(roll_angle, pitch_angle, yaw_angle),  # Quaternion
+            0,  # Body roll rate in radian
+            0,  # Body pitch rate in radian
+            radians(yaw_rate),  # Body yaw rate in radian/second
+            thrust,  # Thrust
+        )
+        self.send_mavlink(msg)
+
+
+    def set_attitude(
+            self,
+            roll_angle=0.0,
+            pitch_angle=0.0,
+            yaw_angle=None,
+            yaw_rate=0.0,
+            use_yaw_rate=False,
+            thrust=0.5,
+            duration=0,
+    ):
+        '''
+        Note that from AC3.3 the message should be re-sent more often than every
+        second, as an ATTITUDE_TARGET order has a timeout of 1s.
+        In AC3.2.1 and earlier the specified attitude persists until it is canceled.
+        The code below should work on either version.
+        Sending the message multiple times is the recommended way.
+        '''
+        self.send_attitude_target(
+            roll_angle, pitch_angle, yaw_angle, yaw_rate, use_yaw_rate, thrust
+        )
```

Comment:
does this block need to exist? I see that this same code is also in the loop below

Reply:
Yeah, this is code from ardupilot, not something I wrote, I noticed this as well, I'll do some testing to see what, if any difference this makes.